cns/restserver/api.go

```diff
@@ -899,6 +902,99 @@ func (service *HTTPRestService) getNetworkContainerByOrchestratorContext(w http.
 	logger.Response(service.Name, getNetworkContainerResponse, returnCode, err)
 }
 
+// getOrRefreshNetworkContainers is to check whether refresh association is needed.
+// If received  "GET": return all NCs in CNS's state file to DNC for follow-up check
+// If received "POST": Store all the NCs (from the request body that client sent) into CNS's state file
+func (service *HTTPRestService) getOrRefreshNetworkContainers(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] getOrRefreshNetworkContainers")
+
+	switch r.Method {
+	case "GET":
```

Comment:
- use constants from `http` package: `http.MethodGet`, `http.MethodPost`
- extract functions specific to the method:
```go
switch r.Method {
case http.MethodGet:
    handleGetNetworkContainers(w, r)
    return
case http.MethodPost:
    handlePostNetworkContainers(w, r)
    return
default:
    // ...
}
```

Reply:
Implementation updated. Extract functions to specific method.